cmake/CodeCoverage.cmake

```diff
@@ -158,7 +158,7 @@ elseif(NOT CMAKE_COMPILER_IS_GNUCXX)
     endif()
 endif()
 
-set(COVERAGE_COMPILER_FLAGS "-g -O0 -fprofile-arcs -ftest-coverage -fcheck=bounds,do,pointer -ffpe-trap=zero,overflow,invalid"
```

Comment:
@mattldawson this is what I changed to make the tests pass

Reply:
And there's a [bug report at gcc](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=108618). Hopefully we can find out if this is a compiler bug or not